tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_cell_data.py

```diff
@@ -26,16 +26,10 @@ def test_enable_cell_data_via_settings_app(self):
 
         # deal with prompt that sometimes appears (on first setting)
         if cell_data_prompt.is_displayed:
+            self.assertTrue(cell_and_data_settings.is_data_toggle_checked)
             # Cell data should not be enabled until we turn it on via the prompt
-            self.assertFalse(cell_and_data_settings.is_data_toggle_checked)
             self.assertFalse(self.data_layer.get_setting('ril.data.enabled'), "Cell data was enabled before responding to the prompt")
             cell_data_prompt.turn_on()
 
-        # Wait for cell data to be turned on
-        self.wait_for_condition(lambda m: cell_and_data_settings.is_data_toggle_checked)
-
         # verify that cell data is now enabled and connected
         self.assertTrue(self.data_layer.is_cell_data_enabled, "Cell data was not enabled via Settings app")
-        self.wait_for_condition(
```

Comment:
Oops! Sorry, I misread that line, I thought it had the same as self.data_layer.is_cell_data_enabled. I'll add it back.